modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ContentAuditLogSupportImpl.java

```diff
@@ -0,0 +1,675 @@
+import org.osgi.service.component.annotations.Reference;
+
+import com.enonic.xp.audit.AuditLogService;
+import com.enonic.xp.audit.AuditLogUri;
+import com.enonic.xp.audit.AuditLogUris;
+import com.enonic.xp.audit.LogAuditLogParams;
+import com.enonic.xp.content.ApplyContentPermissionsParams;
+import com.enonic.xp.content.ApplyContentPermissionsResult;
+import com.enonic.xp.content.Content;
+import com.enonic.xp.content.ContentId;
+import com.enonic.xp.content.ContentIds;
+import com.enonic.xp.content.ContentName;
+import com.enonic.xp.content.ContentPath;
+import com.enonic.xp.content.ContentPaths;
+import com.enonic.xp.content.ContentPublishInfo;
+import com.enonic.xp.content.ContentVersionId;
+import com.enonic.xp.content.Contents;
+import com.enonic.xp.content.CreateContentParams;
+import com.enonic.xp.content.CreateMediaParams;
+import com.enonic.xp.content.DeleteContentParams;
+import com.enonic.xp.content.DeleteContentsResult;
+import com.enonic.xp.content.DuplicateContentParams;
+import com.enonic.xp.content.DuplicateContentsResult;
+import com.enonic.xp.content.MoveContentParams;
+import com.enonic.xp.content.MoveContentsResult;
+import com.enonic.xp.content.PublishContentResult;
+import com.enonic.xp.content.PushContentParams;
+import com.enonic.xp.content.RenameContentParams;
+import com.enonic.xp.content.ReorderChildContentsParams;
+import com.enonic.xp.content.ReorderChildContentsResult;
+import com.enonic.xp.content.SetContentChildOrderParams;
+import com.enonic.xp.content.UndoPendingDeleteContentParams;
+import com.enonic.xp.content.UnpublishContentParams;
+import com.enonic.xp.content.UnpublishContentsResult;
+import com.enonic.xp.content.UpdateContentParams;
+import com.enonic.xp.content.UpdateMediaParams;
+import com.enonic.xp.context.Context;
+import com.enonic.xp.context.ContextAccessor;
+import com.enonic.xp.context.ContextBuilder;
+import com.enonic.xp.data.PropertySet;
+import com.enonic.xp.data.PropertyTree;
+import com.enonic.xp.name.NamePrettyfier;
+import com.enonic.xp.security.RoleKeys;
+import com.enonic.xp.security.acl.AccessControlEntry;
+import com.enonic.xp.security.auth.AuthenticationInfo;
+import com.enonic.xp.site.CreateSiteParams;
+import com.enonic.xp.site.Site;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+@Component(configurationPid = "com.enonic.xp.content")
+public class ContentAuditLogSupportImpl
+    implements ContentAuditLogSupport
+{
+    private static final String SOURCE_CORE_CONTENT = "com.enonic.xp.core-content";
+
+    private final Executor executor;
+
+    private final AuditLogService auditLogService;
+
+    @Activate
+    public ContentAuditLogSupportImpl( final ContentConfig config,
+                                       @Reference(service = ContentAuditLogExecutor.class) final Executor executor,
+                                       @Reference AuditLogService auditLogService )
+    {
+        this.executor = config.auditlog_enabled() ? executor : null;
```

Comment:
Instead of null use `c -> {}`
And remove all null checks `if ( executor != null )`